src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -52,37 +52,57 @@ public async Task PrepareJobAsync(IExecutionContext context, List<ContainerInfo>
-                // }
-                context.JobContext.Services[containerInfo.ContainerNetworkAlias] = service;
+                    if (response.Context.Container.Id != null)
+                    {
+                        context.JobContext.Container["id"] = new StringContextData(response.Context.Container.Id);
+                        jobContainer.ContainerId = response.Context.Container.Id;
+                    }
+                    
+                    if (response.Context.Container.Network != null)
+                    {
+                        context.JobContext.Container["network"] = new StringContextData(response.Context.Container.Network);
+                        jobContainer.ContainerNetwork = response.Context.Container.Network;
+                    }
+                }
+            
+                // TODO: figure out if we need ContainerRuntimePath for anything
+                // var configEnvFormat = "--format \"{{range .Config.Env}}{{println .}}{{end}}\"";
+                // var containerEnv = await _dockerManager.DockerInspect(executionContext, container.ContainerId, configEnvFormat);
+                // container.ContainerRuntimePath = DockerUtil.ParsePathFromConfigEnv(containerEnv);
+
+                if (response.Context.Services != null)
+                {
+                    for (var i = 0; i < response.Context.Services.Count; i++)
+                    {
+                        var container = response.Context.Services[i]; // TODO: Confirm that the order response.Context.Services is the same as serviceContainers
+                        var containerInfo = serviceContainers[i];
+                        containerInfo.ContainerId = container.Id;
+                        containerInfo.ContainerNetwork = container.Network;
+                        var service = new DictionaryContextData()
+                        {
+                            ["id"] = new StringContextData(container.Id),
+                            ["ports"] = new DictionaryContextData(),
+                            ["network"] = new StringContextData(container.Network)
+                        };
+
+                        // TODO: workout port mappings + format
+                        // foreach (var portMapping in containerInfo.UserPortMappings)
+                        // {
+                        //     // TODO: currently the format is ports["80:8080"] = "80:8080", fix this?
+                        //     (service["ports"] as DictionaryContextData)[$"{portMapping.Key}:{portMapping.Value}"] = new StringContextData($"{portMapping.Key}:{portMapping.Value}");
+                        // }
+                        context.JobContext.Services[containerInfo.ContainerNetworkAlias] = service;
+                    }
+                }
+            }
+
+            if ((object)response.State != null)
```

Comment:
If `context.JobContext["hook_state"]` is null, will it fail when we [serialize it](https://github.com/fhammerl/runner-hooks/pull/56/files?diff=split&w=1#diff-a6fa655b45ddbf3a069e7e1ca20d62f8e405d115c2223da1d19c40b9ff24c2baR119)?

Reply:
The `hookState.ToString` part I mean